Invalid handling of `cfg_attr` for crate level features

<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.
-->

I tried this code:

```rust
#![cfg_attr(not(A), feature(decl_macro))]
```

I expected to see this happen: no errors

Instead, this happened:
```rust
test.rs:1:21: error: cfg predicate could not be checked for MetaItemSeq with path of 'feature' - path must be 'all' or 'any'
    1 | #![cfg_attr(not(A), feature(decl_macro))]
      |                     ^~~~~~~
```

### Meta

- What version of Rust GCC were you using: 0ddec9f44d6

--- a/gcc/rust/ast/rust-ast.cc
+++ b/gcc/rust/ast/rust-ast.cc
@@ -4183,7 +4183,8 @@ Attribute::check_cfg_predicate (const Session &session) const
   if (!is_parsed_to_meta_item ())
     return false;
 
-  return attr_input->check_cfg_predicate (session);
+  auto &meta_item = static_cast<AttrInputMetaItemContainer &> (*attr_input);
+  return meta_item.get_items ().front ()->check_cfg_predicate (session);
 }
 
 std::vector<Attribute>

--- a/gcc/testsuite/rust/compile/cfg-attr.rs
+++ b/gcc/testsuite/rust/compile/cfg-attr.rs
@@ -1,0 +1,7 @@
+// { dg-additional-options "-w -frust-cfg=A" }
+#![cfg_attr(not(B), allow(dead_code))]
+#![cfg_attr(A, allow(while_true))]
+
+#[cfg_attr(not(B), allow(deprecated))]
+#[cfg_attr(A, allow(drop_bounds))]
+fn main() { }

